### PR TITLE
Migrate from CircleCI to GitHub Actions

### PR DESCRIPTION
**Pull request by kzkn**

### CODE REVIEW

**Comment by kzkn** [comment]
@gingermusketeer 
I tried to continue to use CircleCI (#282), but my commits were not able to run CI due to [errors](https://app.circleci.com/pipelines/github/ruby-formatter/rufo/308/workflows/3a006649-c304-4163-b5f0-97e7823cb8bd/jobs/1302). Perhaps something is under configured, but I figured it would be faster to move to GitHub Actions than to investigate the details.
What do you think?

**Comment by gingermusketeer** [comment]
Feel free to do what you think is best @kzkn !

**Comment by gingermusketeer** [comment]
And you don’t need my approval to merge as I will not be able to respond too quickly

**Comment by tnir** [comment]
@kzkn Due to January 4, 2023 security incident https://circleci.com/blog/jan-4-2023-incident-report/, the existing integration between GitHub and CircleCI was revoked by CircleCI. I prefer GitHub Actions to CircleCI as we can avoid such a credential leak!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - uses: actions/checkout@v3
+
+    - name: Setup Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby_version }}
+
+    - name: Bundle install
+      run: |
+        bundle check || bundle install
```
Comment (tnir) [inline .github/workflows/ci.yml:28]: Why not using `bundler-cache` on `ruby/setup-ruby` action?

```suggestion
      uses: ruby/setup-ruby@v1
      with:
        ruby-version: ${{ matrix.ruby_version }}
        bundler-cache: true
```

**Review by tnir — COMMENTED** [review]
@kzkn Great! One tiny comment.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kzkn) [reply .github/workflows/ci.yml:28]: I just forgot it. Thanks!

**Comment by kzkn** [comment]
@gingermusketeer @tnir thanks for the comment :smile: 
The integration with CircleCI will be removed and migrated to GitHub Actions.

**Pull request merged**